nibabel/imagestats.py

```diff
@@ -0,0 +1,58 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+Functions for computing image statistics
+"""
+
+import numpy as np
+
+
+def mask_volume(img, units='mm3'):
```

Comment:
Thinking about this a bit, if `units=="vox"`, then the output should be integer; there's no good reason to have a float for counting non-zero voxels. That makes the types a bit messy. What if we did (simplified):

```Python
def count_nonzero_voxels(img):
    return np.count_nonzero(img.dataobj)

def mask_volume(img):
    nz_vox = count_nonzero_voxels(img)
    return np.prod(img.header.get_zooms()[:3]) * nz_vox
```

The conditional logic inside this function basically evaporates, and is moved into `cmdline.stats.main` to select which function to call.

Reply:
Sounds good to me. I have simply left the code a bit more verbose for better readability.